File: src/main/java/com/fury/game/entity/combat/scripts/glacor/GlacyteCombat.java
package com.fury.game.entity.combat.scripts.glacor;

import com.fury.game.content.skill.Skill;
import com.fury.core.model.node.entity.actor.figure.Figure;
import com.fury.core.model.node.entity.actor.figure.mob.Mob;
import com.fury.core.model.node.entity.actor.figure.player.Player;
import com.fury.game.entity.combat.CombatScript;
import com.fury.game.npc.bosses.glacors.Glacyte;
import com.fury.game.system.files.loaders.npc.MobCombatDefinitions;
import com.fury.game.world.update.flag.block.Animation;

public class GlacyteCombat extends CombatScript {

    @Override
    public Object[] getKeys() {
        return new Object[] { 14303 };
    }

    @Override
    public int attack(final Mob mob, final Figure target) {
        Glacyte glacyte = (Glacyte) mob;
        MobCombatDefinitions defs = mob.getCombatDefinition();
        if (target.isPlayer()) {
            Player player = (Player) target;
            if (glacyte.getEffect() == 1)
                player.getSkills().drain(Skill.PRAYER, 20);//should be only on hit
        }
        mob.perform(new Animation(defs.getAttackAnim()));
        delayHit(mob, 0, target, getMeleeHit(mob, getRandomMaxHit(mob, defs.getMaxHit(), target)));
        return mob.getCombat().getAttackSpeed();
    }

}

File: src/main/kotlin/com/fury/game/node/entity/mob/combat/GlacyteCombat.kt
package com.fury.game.node.entity.mob.combat

import com.fury.core.model.node.entity.actor.figure.Figure
import com.fury.game.entity.character.combat.Hit
import com.fury.game.node.entity.actor.figure.mob.MobCombat
import com.fury.game.npc.bosses.glacors.Glacyte
import com.fury.util.Misc

class GlacyteCombat(private val mob: Glacyte): MobCombat(mob) {

    override fun handleHit(hit: Hit) {
        if (mob.targetIndex == -1)
            mob.targetIndex = hit.source.index
        var damage = hit.damage
        if (mob.effect.toInt() == 2) {
            val target = mob.mobCombat!!.target
            if (target != null && damage > 0)
                damage = (((6 - Misc.getDistance(target, mob.glacor)) / 10 + .4) * damage).toInt()
            hit.damage = damage
        }
        super.handleHit(hit)
    }

    override fun sendDeath(source: Figure?) {
        mob.glacor!!.removeGlacyte(mob)
        super.sendDeath(source)
    }
}